### PR TITLE
Properly handle query params alongside request bodies

### LINKED ISSUE #29: Bad request (400) when using the search API
I am using 0.4.0 and I am trying to use the `search/stories` endpoint, but I am getting a 400 when calling the endpoint with clubhouse-lib. If I use curl or my own python lib (based on requests) everything works fine. The only difference I can see from the request is that the `Content-Length` header is not set by clubhouse-lib. I might be a red herring. 

`client.getResource('search/stories', { query: 'owner:jm' }).then(console.log).catch(console.log);`

If I can get it working I'll make a PR for it, but for now it's not working. :\

### PR DESCRIPTION
**Pull request by eliperkins**
This varies behavior for GET requests versus all other requests, where parameters
for GET requests should be in the query string, whereas other methods should use
the body.

This adds in the [`query-string`](https://www.npmjs.com/package/query-string) module to help us create these queries.

Fixes #29

### CODE REVIEW

**Comment by eliperkins** [comment]
@j-martin any chance you'd like to review this PR? I know you mentioned that `requests` in Python didn't seem to exhibit the same behavior, but testing this branch locally with the REPL seems to allow the query you were attempting to run!


```shell
~/src/clubhouse/clubhouse-lib eli/fix-29*
❯ node --experimental-repl-await
> const Client = require('./');
undefined
> const client = Client.create('5c59f82e-8c22-4af2-854b-a91f52103671');
undefined
> await client.getResource('search/stories', { query: 'owner:eliperkins' });
{ next:
   '/api/beta/search/stories?query=owner%3Aeliperkins&next=4fb8da73a4b90eef85eab94345caedf423cef770~9',
  data:
   [ { entity_type: 'story',
       archived: false,
       created_at: '2018-12-06T17:41:56Z',
       updated_at: '2019-02-12T15:38:53Z',
       id: 48240,
       external_id: null,
...
```

File: src/TokenRequestFactory.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 /* @flow */
 
+import { stringify } from 'query-string';
+
 import type { RequestFactory } from './types';
 
+require('fetch-everywhere');
```
Comment (jeremyheiler) [inline src/TokenRequestFactory.js]: Curious: why `require` instead of `import`?

Comment (eliperkins) [reply src/TokenRequestFactory.js]: Because `fetch-everywhere` executes side-effects by adding `global`s when requiring it: https://github.com/lucasfeliciano/fetch-everywhere/blob/b472faedba57cff3d5a6d1f2183812ab47a9f387/fetch-npm-node.js#L11-L16

Using `import` here would/could be tree-shaken or lazily-required, meaning we could execute our code before the side-effect occurred.

tl;dr: side-effects

Comment (jeremyheiler) [reply src/TokenRequestFactory.js]: Thanks!

**Comment by j-martin** [comment]
@eliperkins yep, that fixed my issue with the search endpoint. Thanks!
